### ISSUE TITLE
Bump Version Request

### ISSUE THREAD

**Issue by DerDennisOP** [YannickJadoul/Parselmouth#130]
I coudn't get the version 0.4.4 to be build on nixos, due to too old versions beeing used. (The latest commit works for me tho) Please bump the version to something stable. I would really like to add this project to the NixOS/nixpkgs repository.

**Comment by YannickJadoul** [comment]
Please give me more details. Why is the build not working, which version is too old?
The `stable` branch is working on CI, on modern GitHub action runners which are supported, and is working with Python 3.12. So I will need more details.

> I would really like to add this project to the NixOS/nixpkgs repository.

Prebuilt binary wheels for the main distributions are on PyPI, the official Python repo. Don't they work? Why does it need to be added to a separate repository and create confusion and an unofficial distribution?

**Comment by DerDennisOP** [comment]
I'm not sure, what exactly is broken. I had to deactivate X11 and then I got some errors that stated: `no type named 'uint16_t' in namespace 'std'`
So I switched from tag v0.4.4 to the latest commit on master branch, which then worked.

> Prebuilt binary wheels for the main distributions are on PyPI, the official Python repo. Don't they work? Why does it need to be added to a separate repository and create confusion and an unofficial distribution?

I'm using a specific linux distribution, which wants everything reproducible. That means no prebuild binaries, but building it's own binaries. So if I were to use arm64 architecture it would still work and just build binaries that aren't prebuild.
You can check all the packges of the distro here: https://search.nixos.org/packages . On NixOS there is basically also no other packagemanager. You are stuck with nix. This leads to some advantages: like you just have to use one package manager and everything stays reproducible. Obvly there are some disadvantages (like you are stuck with nix).

All in all I'm requesting a new version because it worked for me better with the latest commit. It's nothing urgent, but I can't really merge it into NixOS/nixpkgs repo without it beeing an official release.

**Comment by YannickJadoul** [comment]
> I'm not sure, what exactly is broken.

Still, any build logs, or anything? Compiler versions?
It could be something with a newer version of GCC actually, so knowing about how exactly you're building stuff would obviously help.
It would help to diagnose and to maybe notice that it's a broader problem. E.g., if it's indeed this newer version of GCC, that means that the current sdists on PyPI can also not be built by people with a newer GCC compiler...

> So I switched from tag v0.4.4 to the latest commit on master branch, which then worked.

Well, the `master` branch is in development and not ready to be released, at this point. So that's why I have the `stable` branch, for bugfixes and patch releases while new functionality gets added on `master`. I am working on a new release with Praat updates, etc, though, but I cannot make any promises there on when that will get released.

Anyway, I've just cherry-picked a couple of commits which I suspect might help with this, onto `stable`. So feel free to give those a shot. I know, it's not yet a tagged release, but it's closest to the current latest release, and probably the fastest to get a new patch release.

> I'm using a specific linux distribution, which wants everything reproducible. That means no prebuild binaries (you can add prebuild binaries but it's not really wanned), but building it's own binaries.

Alright, I see, thanks for the explanation! I still don't fully understand, cause Python, `pip`, and PyPI have source builds and options to not download pre-built binaries, of course, but I guess it makes sense in a certain context.
Who normally maintains these nix packages? Because at any rate, I don't currently have the bandwidth to support any packaging beyond the standard Python package manager and PyPI. So I wouldn't like any sort of unofficial package to get inconsistent or out of date with the main distributed one.

**Comment by DerDennisOP** [comment]
> Who normally maintains these nix packages? Because at any rate, I don't currently have the bandwidth to support any packaging beyond the standard Python package manager and PyPI. So I wouldn't like any sort of unofficial package to get inconsistent or out of date with the main distributed one.

Well, for this specific Project I'll be the maintainer. https://github.com/NixOS/nixpkgs/commit/b19b62fc6500323a82a46719f8bba464b9cbf7a0
The Package Update is done by a Bot, but when there are dependency / build changes the maintainer has to update it.
The Nixpkgs repo is in the top 10 of githubs biggest repos. If there is an update necessary you can open up an Issue on Nixpkgs and tag me there.

Here is the CMakeCache.txt: https://gist.github.com/DerDennisOP/f8b1d9925bbd14b89c8e7f04bc0c8dcb

As an explanation about Nix: Please don't be off put by the the weired /nix/store/[weired hash]* paths. Basically in nix you store every program / python dep / config file / etc. in the /nix/store. every program is then symlinked to the right place by nix. This is like using python venv, but for your entire system. You could switch all your programs within a second, and afterwards switch back without breaking anything.

**Comment by YannickJadoul** [comment]
Thanks for all the extra explanation, @DerDennisOP! That does help!

Could you confirm that the current `stable` branch also works? That would mean that the two cherry-picked commits fix the source build.

If so, I will still look at whether the current version also works with Python 3.13 and build wheels, and then I could release a new 0.4.5, while working towards a full new 0.5.0 version.

**Comment by DerDennisOP** [comment]
The stable version (commit: 5bbb97a6a33778746b67cd07c084fe5ae91ac437) does work for me.

**Comment by DerDennisOP** [comment]
Please check this please:
```
ImportError while loading conftest '/build/source/tests/conftest.py'.
tests/conftest.py:53: in <module>
    from resource_fixtures import *
tests/resource_fixtures.py:19: in <module>
    import pytest_lazyfixture
E   ModuleNotFoundError: No module named 'pytest_lazyfixture'
```

**Comment by DerDennisOP** [comment]
Ok, I think it was because I used pytest-lazy-fixtures instead of -fixture. I'm currently also testing Python 3.13

**Comment by YannickJadoul** [comment]
> I had to install the "future" module. please include it in your requirements.txt

It's there, in `tests/requirements.txt`

**Comment by DerDennisOP** [comment]
ok, well then I somehow missed it

**Comment by DerDennisOP** [comment]
https://github.com/NixOS/nixpkgs/pull/334636 I got this through. I'll close this request.

**Comment by YannickJadoul** [comment]
@DerDennisOP, I just released 0.4.5 after all, containing these changes (and with wheels for Python 3.13): https://github.com/YannickJadoul/Parselmouth/releases/tag/v0.4.5
If you want to update to an actual release version, it's finally there now!
